dspace-server-webapp/src/main/java/org/dspace/app/rest/AccessStatusRestController.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.dspace.access.status.service.AccessStatusService;
+import org.dspace.app.rest.model.AccessStatusRest;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.authorize.service.ResourcePolicyService;
+import org.dspace.content.Item;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.discovery.SearchServiceException;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.TemplateVariable;
+import org.springframework.hateoas.TemplateVariable.VariableType;
+import org.springframework.hateoas.TemplateVariables;
+import org.springframework.hateoas.UriTemplate;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This is an utility endpoint to calculate the access status of an item.
+ */
+@RestController
+@RequestMapping("/api/" + AccessStatusRestController.CATEGORY)
+public class AccessStatusRestController implements InitializingBean {
```

Comment:
@nibou230 : I haven't had a chance to give this a full review yet, but I wanted to immediately note that this endpoint design looks slightly odd to me because you've created an *Item-specific* endpoint outside of the `/api/core/items` existing endpoint.

In the case of an Item "accessStatus", I would have expected it to be available from the Item resource itself, e.g. at `/api/core/items/{uuid}/accessStatus`.  This would be similar to how you can retrieve an Item's primary Thumbnail from `/api/core/items/{uuid}/thumbnail` and get other properties of an Item in the same manner.  This makes it very easy for the UI to retrieve the `accessStatus` at the same time it retrieves the Item -- without requiring multiple calls to the REST API (one to grab the item and then one to get its accessStatus). This is also a more RESTful design for the API...it keeps all the Item-specific properties under the Item endpoint.

The way this is usually achieved is by using a `*LinkRepository` class (instead of a `@RestController`).  For example, `/api/core/items/{uuid}/thumbnail` is controlled by the ItemThumbnailLinkRepository: https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ItemThumbnailLinkRepository.java   So, this class might be rethought as an `ItemAccessStatusLinkRepository`

Apologies if this wasn't clear from the codebase itself.  This is one of the reasons though that we recommend creating a REST Contract early on (even while you are just starting development) -- it provides a way for us to immediately provide feedback on the design of new endpoints.

Glad to answer questions on this & sorry if I somehow glossed over creating a REST Contract for feedback purposes.

Reply:
Hi @tdonohue, I'm back from the long Easter weekend.

I did exactly what you mentionned at first, but I reverted it back because I thought those `*LinkRepository` endpoints needed to expose an existing object in the database, let's say a "physical" object. I know this status could be enventually stored somewhere to avoid recalculating it each time, so it make sense to think ahead of time and have a correct endpoint right away.

Oh well, it seems I should have followed my first guess.

I will adjust the 3 PR accordingly.